app/controllers/health_check_controller.rb

```diff
@@ -0,0 +1,21 @@
+class HealthCheckController < ApplicationController
```

Comment:
should we rate limit these? :D 

Aside from rate limiting, whose default could be fine, should we use some sort of authentication method? Basic authentication with a server side key is fine. I'm pretty sure any "pinger" out there can use basic auth, even `curl` can. 

I'm a bit wary of the idea that third parties can check this endpoint at random, especially the one about search. Why should they know if Elasticsearch is up or not?

Something like:

```ruby
class HealthCheckController < ApplicationController
   http_basic_authenticate_with name: AppConfig["HEALTH_CHECK_AUTH_USERNAME"], password: AppConfig["HEALTH_CHECK_AUTH_PASSWORD"], except: :index
end
```

and then:

```shell
$ curl -u username:password https://dev.to/health-checks/app
```

Last but not least, what if we use the "canary healtheck" strategy ? Instead of 3 endpoints, we build one, which returns a structured JSON with all the results, something like:

```json
{
  "app": {
    "success": true
  },
  "db": {
    "success": true
  },
  "search": {
    "success": true
  }
}
```

this way metrics can be added in the future without re-doing the client (which, if not updated will just keep checking true or false) and we'd have one endpoint

ps. have you considered making this an actual API controller?

Reply:
> should we rate limit these? :D

Nah I think our defaults are sufficient

> have you considered making this an actual API controller?

Yeah I considered making an API controller with authentication but then did some testing and figured out that we did not authenticate these endpoints at Kenna. I couldn't remember if there was a reason for that or not so I stuck with that approach in case there was a reason. After chatting with @jdoss he would rather have them authenticated so I will update that. 

> what if we use the "canary healtheck" strategy

I think having separate ones is more explicit and allows us more flexibility on our checks so once the db comes back as up we can stop checking that, etc. This could totally change down the road but I like separate endpoints to start.

Reply:
>  I think having separate ones is more explicit and allows us more flexibility on our checks so once the db comes back as up we can stop checking that, etc. This could totally change down the road but I like separate endpoints to start.

I was thinking that the main limitation with this is that if the app is down, both ES and the DB will be flagged as down as these checks both depend on Rails to be up to work.

I guess that's why I had the idea of consolidating them in one single endpoint